src/components/Header/Header.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { selectUserData } from "../../redux/user/selectors";
+import { signOut } from "../../redux/user/operations";
+import CustomModal from "../CustomModal/CustomModal";
+import LogoutForm from "../LogoutForm/LogoutForm";
+import { useAppDispatch, useAppSelector } from "../../redux/hooks";
+import "../../img/icons.svg";
+import s from "./Header.module.css";
+import Logo from "../Logo/Logo";
+import { Icon } from "../Icon/Icon";
+import { useMediaQuery } from "react-responsive";
+import clsx from "clsx";
+
+const Header = () => {
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const userData = useAppSelector(selectUserData);
+
+  const closeModal = () => {
+    setModalIsOpen(false);
+  };
+
+  const openModal = () => {
+    setModalIsOpen(true);
+  };
+
+  const handleLogout = async () => {
+    try {
+      await dispatch(signOut());
+      navigate("/login");
```

Comment:
user will be redirected to the login according to the private routes logic, no need to make additional navigation